### PR TITLE
#3024 fix set eto start date widget

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this PR fixes all points from the #3024 except the last one.
the last one is a bug in the "react-datetime". THere's no quick fix for it, so our options are

- fork "react-datetime" and fix it
- wait till they fix it (can be long, there's not so much activity on the repo)
- write our own widget (could be the best approach because "react-daytime" has ux issues, but needs design and some time)

this point will be handled in a separate ticket anyway

### CODE REVIEW

**Comment by AlexeyTuboltsev** [comment]
1555

**Comment by AlexeyTuboltsev** [comment]
1426

File: app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts [hunk 1]
```diff
@@ -46,14 +47,18 @@ describe.skip("Eto start date setup", () => {
...
       // should not be allowed to set a date that is too soon
-      const falseDate = newStartDate.clone().subtract(17, "days");
+      const falseDate = newStartDate
+        .clone()
+        .subtract(19, "days")
+        .subtract(8, "hours");
 
       cy.get(tid("eto-settings-start-date-open-date-picker"))
-        .wait(5000)
+        // .wait(5000)
```
Comment (Moejoe90) [inline app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts]: Please remove

File: app/test-e2e/eto-view/EtoViewForInvestor.spec.e2e.io.ts [hunk 2]
```diff
@@ -222,7 +222,7 @@ describe("Eto Investor View", () => {
   });
 
   describe("Fixtures tests", () => {
-    it("coming soon state should have pitch deck", () => {
+    it.skip("coming soon state should have pitch deck", () => {
```
Comment (Moejoe90) [inline app/test-e2e/eto-view/EtoViewForInvestor.spec.e2e.io.ts]: why are we skipping this?

File: app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts [hunk 3]
```diff
@@ -46,14 +47,18 @@ describe.skip("Eto start date setup", () => {
             `${weekdayUTC(newStartDate.toDate())}, ${utcTime(newStartDate.toDate())}`,
           ),
         )
+        .wait(5000)
```
Comment (Moejoe90) [inline app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts]: do we need to wait 5000 ms

File: app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts [hunk 4]
```diff
@@ -6,16 +6,17 @@ import { closeModal, confirmAccessModal } from "../utils";
...
+describe("Eto start date setup", () => {
   // TODO: Enable E2E test
   it("sets the date", () => {
     loginFixtureAccount("ISSUER_SETUP", {
       kyc: "business",
     }).then(() => {
       const newStartDate = moment
         .utc()
-        .startOf("day")
+        // .startOf("day")
```
Comment (Moejoe90) [inline app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts]: commented code

File: app/modules/tx/transactions/sagas.ts [hunk 5]
```diff
@@ -125,15 +125,16 @@ export function* investorPayoutRedistributeSaga(
...
   try {
+    yield put(actions.etoFlow.setEtoDateStart());
+
     yield txSendSaga({
       type: ETxSenderType.ETO_SET_DATE,
       transactionFlowGenerator: etoSetDateGenerator,
     });
     logger.info("Setting ETO date successful");
-    // cleanup & refresh eto data
-    yield put(actions.etoFlow.cleanupStartDate());
```
Comment (Moejoe90) [inline app/modules/tx/transactions/sagas.ts]: so now you don't use 
```
    yield put(actions.etoFlow.cleanupStartDate());	
``` 
anymore?

File: app/modules/eto-flow/sagas.ts [hunk 6]
```diff
@@ -189,8 +195,13 @@ function* startSetDateTX(_: TGlobalDependencies, action: TAction): any {
 }
 
 export function* cleanupSetDateTX(): any {
-  const eto = yield select(selectIssuerEto);
-  yield put(actions.eto.loadEto(eto.etoId));
+  const state: IAppState = yield select();
+  const newStartDate = selectNewPreEtoStartDate(state);
```
Comment (Moejoe90) [inline app/modules/eto-flow/sagas.ts]: please use this approach 
```
  const newStartDate = yield select(selectNewPreEtoStartDate);	
```
also for `newStartDate` for consistancy i know you want to save typings but we had a discussion about this before and agreed to keep this style

File: app/test-e2e/eto-view/EtoViewForInvestor.spec.e2e.io.ts (hunk 2 above)
Comment (AlexeyTuboltsev) [reply app/test-e2e/eto-view/EtoViewForInvestor.spec.e2e.io.ts]: ups

File: app/modules/tx/transactions/sagas.ts (hunk 5 above)
Comment (AlexeyTuboltsev) [reply app/modules/tx/transactions/sagas.ts]: yes, it's being done from component

File: app/modules/eto-flow/sagas.ts (hunk 6 above)
Comment (AlexeyTuboltsev) [reply app/modules/eto-flow/sagas.ts]: done

File: app/components/eto/dashboard/chooseEtoStartDateWidget/ChooseEtoStartDateWidget.tsx [hunk 7]
```diff
@@ -323,20 +333,33 @@ const SetNewDate: React.ComponentType<IStateProps & IDispatchProps> = props => (
...
-    columnSpan={props.columnSpan}
-  >
+  <WidgetPanel columnSpan={props.columnSpan}>
     <div className={styles.content}>
       <p className={styles.text}>
-        <FormattedMessage id="settings.choose-pre-eto-date.book-building-will-stop" />
+        <FormattedHTMLMessage
+          tagName="span"
+          id="settings.choose-pre-eto-date.book-building-will-stop"
+          values={{ minOffsetPeriod: props.minOffsetPeriod.div(60 * 60 * 24).toNumber() }}
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/eto/dashboard/chooseEtoStartDateWidget/ChooseEtoStartDateWidget.tsx]: done

File: app/components/eto/dashboard/bookBuildingWidget/BookBuildingWidget.tsx [hunk 8]
```diff
@@ -252,4 +278,5 @@ export const BookBuildingWidget = compose<React.FunctionComponent<IExternalProps
       }
     },
   }),
+  branch<IStateProps>(props => !props.bookBuildingStats, renderComponent(WidgetLoading)),
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/eto/dashboard/bookBuildingWidget/BookBuildingWidget.tsx]: yes there's even a ticket for it. #3042

File: .circleci/config.yml [hunk 9]
```diff
@@ -8,7 +8,7 @@ jobs:
 
     steps:
       - checkout
-    
+
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply .circleci/config.yml]: i'm very sorry 😱

File: app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts (hunk 3 above)
Comment (AlexeyTuboltsev) [reply app/test-e2e/eto-registration/SetEtoStartDate.spec.e2e.io.ts]: removed

**Comment by rudolfix** [comment]
@AlexeyTuboltsev many conflicts here - was there any other branch that modified start date widget?

**Pull request merged**